app/middlewares/sessions.js

```diff
@@ -0,0 +1,21 @@
+const { verifyToken } = require('../services/sessions');
+const errors = require('../errors');
+const { INVALID_TOKEN } = require('../../config/constants');
+
+exports.authenticateSession = async (req, res, next) => {
+  try {
+    const bearerHeader = req.headers && req.headers.authorization;
+    if (bearerHeader) {
+      const token = bearerHeader.split(' ')[1];
```

Comment:
With this approach you're accepting any authorization header, and not only Bearer. For security reasons you shouldn't accept any header that does not start with Bearer.

Reply:
maybe `if (authHeader.startsWith("Bearer ") token = authHeader.substring(7, authHeader.length);`